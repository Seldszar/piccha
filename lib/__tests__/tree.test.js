const Tree = require("../tree");

describe("Tree", () => {
  let tree;

  test("should create an empty tree", () => {
    tree = new Tree();

    expect(tree).toMatchSnapshot();
  });

  test("should return a node", () => {
    tree = new Tree({
      "/lorem": "ipsum",
    });

    expect(tree.get("/lorem")).toMatchSnapshot();
  });

  test("should add a node", () => {
    tree = new Tree();

    tree.set("/ipsum", "dolor");

    expect(tree).toMatchSnapshot();
  });

  test("should delete an existent node", () => {
    tree = new Tree({
      "/lorem": "ipsum",
    });

    expect(tree.delete("/lorem")).toMatchSnapshot();
  });

  test("should delete a non-existent node", () => {
    tree = new Tree();

    expect(tree.delete("/lorem")).toMatchSnapshot();
  });

  test("should merge values into the tree", () => {
    tree = new Tree({
      "/lorem": "ipsum",
    });

    tree.merge({
      "/lorem": undefined,
      "/ipsum": "dolor",
    });

    expect(tree).toMatchSnapshot();
  });

  test("should emit a change event", async () => {
    tree = new Tree();

    const promise = new Promise(resolve => {
      tree.on("change", resolve);
    });

    tree.set("/lorem", "ipsum");

    await expect(promise).resolves.toMatchSnapshot();
  });

  test("should return the string representation of the tree", () => {
    expect(tree.toString()).toMatchSnapshot();
  });
});
