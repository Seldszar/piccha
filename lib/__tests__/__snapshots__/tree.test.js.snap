// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tree should add a node 1`] = `
Object {
  "root": Object {
    "children": Map {
      "ipsum" => Object {
        "children": Map {},
        "path": "/ipsum",
        "value": "dolor",
      },
    },
    "path": "/",
    "value": undefined,
  },
}
`;

exports[`Tree should create an empty tree 1`] = `
Object {
  "root": Object {
    "children": Map {},
    "path": "/",
    "value": undefined,
  },
}
`;

exports[`Tree should delete a non-existent node 1`] = `false`;

exports[`Tree should delete an existent node 1`] = `true`;

exports[`Tree should emit a change event 1`] = `
Object {
  "newValue": "ipsum",
  "oldValue": undefined,
  "path": "/lorem",
}
`;

exports[`Tree should merge values into the tree 1`] = `
Object {
  "root": Object {
    "children": Map {
      "ipsum" => Object {
        "children": Map {},
        "path": "/ipsum",
        "value": "dolor",
      },
    },
    "path": "/",
    "value": undefined,
  },
}
`;

exports[`Tree should return a node 1`] = `
Object {
  "children": Map {},
  "path": "/lorem",
  "value": "ipsum",
}
`;

exports[`Tree should return the string representation of the tree 1`] = `"[object Tree]"`;
